dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifGoldenCopyIngestor.kt

```diff
@@ -94,6 +107,31 @@ class GleifGoldenCopyIngestor(
         logger.info("Finished processing of file $csvFile in ${getExecutionTime(start)}.")
     }
 
+    private fun prepareMappingFile() {
+        logger.info("Starting LEI-ISIN mapping update cycle for latest file.")
+        val tempFile = File.createTempFile("gleif_mapping_update", ".csv")
+        processMappingFile(tempFile, gleifApiAccessor::getIsinMappingFile)
+    }
+
+    @Synchronized
```

Comment:
@SiegfriedSobkowiak do I understand correctly that we use the `@Synchronized` annotation here (and above for the gleif diff) to prevent this from running more than once at the same time, e.g. if last sundays task is not finished yet, do not start the next update yet, as an extreme case?